packages/react-native/ReactCommon/react/runtime/TimerManager.cpp

```diff
@@ -344,8 +357,7 @@ void TimerManager::attachGlobals(jsi::Runtime& runtime) {
                   rt, "The first argument to setInterval must be a function.");
             }
             auto callback = args[0].getObject(rt).getFunction(rt);
-            auto delay =
-                count > 1 && args[1].isNumber() ? args[1].getNumber() : 0;
+            auto delay = count > 1 ? coerceNumberTimeout(rt, jsi::Value { rt, args[1] }) : 0.0;
```

Comment:
There should be no need to create a new jsi::Value here
```suggestion
            auto delay = count > 1 ? coerceNumberTimeout(rt, args[1]) : 0.0;
```